packages/inter-protocol/src/proposals/add-auction.js

```diff
@@ -57,6 +67,16 @@ export const addAuction = async ({
     legacyKitP,
   ]);
 
+  await E.when(
+    installationP,
+    installation =>
+      E(E(agoricNamesAdmin).lookupAdmin('installation')).update(
+        'auctioneer',
+        installation,
+      ),
+    err => console.error(`🚨 failed to update vaultFactory installation`, err),
+  );
```

Comment:
Using the promise space is supposed to take care of agoricNames for you:

```suggestion
  produceInstallation.reset(installation);
  produceInstallation.resolve(installation);
```

combined with:

```js
    installation: {
      produce: { auction: produceInstallation },
    },
```

I'm not 100% sure these are equivalent. thinking it over.

Reply:
Reviewing the code for `makePromiseSpaceForNameHub`, I see that
  - using the promise space updates agoricNames as a side-effect, so they stay in sync, but
  - updating agoricNames directly does _not _ update the promise space, so if someone later tries to use `installation.consume.auctioneer`, they'll get the old installation

Reply:
I will fix the code in upgradeVaults as well.

Reply:
This turned into a major clean-up.

@dckc  this means `upgrade-vaults.js` needs a fresh review.